tests/benchmark_utf8.cpp

```diff
@@ -46,8 +45,8 @@ char *randomString(int len) {
   char *s = new char[len+1];
   memcpy(s, utf8_corpus + startIdx, len);
   s[len] = '\0';
-  while(((unsigned char)s[len-1] & 0xC0) == 0x80) { s[--len] = '\0'; }
-  while(((unsigned char)s[len-1] & 0xC0) == 0xC0) { s[--len] = '\0'; }
+  while(len > 0 && ((unsigned char)s[len-1] & 0xC0) == 0x80) { s[--len] = '\0'; }
+  while(len > 0 && ((unsigned char)s[len-1] & 0xC0) == 0xC0) { s[--len] = '\0'; }
   assert(len >= 0);
```

Comment:
Shouldn't this assert have fired if len ended up doing below zero?  (i.e. why didn't the non-asan build fail too).

Reply:
Isn't it pretty bad that we call `srand(time(NULL));` below?   That could cause the test to fail depending on the time of day :)

Reply:
The asan error doesn't happen in the body of the loop, but in the condition. That is, `s=0` and we looked at address `s[0-1]`, and then decided not to decrement `s` any further. But by looking at `s[-1]` asan errors.

Reply:
Thanks, that makes sense now.